Manager Implemented, and edited

FinagleLord opened the pull request.

FinagleLord committed:
Wrapper Added - Bug Fixes - UnFlattened

Zayen-X commented on zaps/contracts/OlympusZap.sol:
```diff
@@ -0,0 +1,246 @@
+        address swapTarget,
+        bytes calldata swapData,
+        address affiliate,
+        uint maxBondPrice,
```
How do we get this off-chain? this is an onchain request anyway.

Zayen-X commented on zaps/contracts/OlympusZap.sol:
```diff
@@ -0,0 +1,246 @@
+    ) external payable stopInEmergency returns ( uint OHMRec ) {
+        if ( bond ) {
+            // make sure market exists for given principal/toToken
+            require(principalToDepository[ toToken ] != address(0), "bonding market doesn't exist");
```
If the adapter manages this data we can get maxPayout from it as well.

Zayen-X commented on zaps/contracts/OlympusZap.sol:
```diff
@@ -0,0 +1,246 @@
+            uint256 tokensBought = _fillQuote(fromToken, toToken, toInvest, swapTarget, swapData);
+            require(tokensBought >= minToToken, "High Slippage");
+            // buy bond on the behalf of user
+            IBondDepository( principalToDepository[ toToken ] ).deposit(tokensBought, maxBondPrice, msg.sender);
```
This line would be calling the adapter.

Zayen-X reviewed (changes requested):
The adapter doesn't seem to make sense.

FinagleLord committed:
Update OlympusZapManager.sol

FinagleLord committed:
Update OlympusZap.sol

Commit:
Smol fixes

- Add OlympusDAO author
- Minor comment changes
- Remove zapInBond event as the type of zap can be inferred from toToken offchain
- Add sushi/uni Zaps as pre-approved targets
- Fix casting in _enterOlympus function
- Add olympusZapManager setter which is restricted to onlyOlympusZapManagerOwner

FinagleLord committed:
Merge pull request #1 from suhailgme/main

Smol fixes

FinagleLord committed:
add Suhails changes

FinagleLord committed:
Update OlympusZapManager.sol

Zayen-X commented on zaps/contracts/OlympusZapManager.sol:
```diff
@@ -0,0 +1,70 @@
+        // make sure market exists for given principal/toToken
+        require( principalToDepository[ _principal ] != address(0), "bonding market doesn't exist");
+        // buy bond on the behalf of user
+        depository.deposit( _amount, _maxBondPrice, msg.sender );
```
Err the only thing here, is that msg.sender is going to be the zap contract. the zap contract is gonna be mega rich with Ohm. lol. 

You need to pass the `_depositor` address down from the Zap Method

Zayen-X replied:
Also called out by @suhailgme we need to approve the deposit to the bond depository. Its going to fail otherwise.

Zayen-X reviewed (changes requested):
Minor thing about msg.sender.

Zayen-X commented on zaps/contracts/OlympusZapManager.sol:
```diff
@@ -0,0 +1,70 @@
+    ) external returns ( uint ) {
+        IBondDepository depository = IBondDepository( principalToDepository[ _principal ] );
+        // make sure market exists for given principal/toToken
+        require( principalToDepository[ _principal ] != address(0), "bonding market doesn't exist");
```
We can just cast depository to an address to save gas for reads `address(depository)`

Zayen-X commented on zaps/contracts/OlympusZap.sol:
```diff
@@ -0,0 +1,232 @@
+        require(fromToken == olympusZapManager.sOHM() || fromToken == olympusZapManager.wsOHM(), "fromToken must be sOHM or wsOHM");
+        amountIn = _pullTokens( fromToken, amountIn );
+        uint256 OHMRec = _exitOlympus( fromToken, amountIn );
+        tokensRec = _fillQuote( olympusZapManager.OHM(), toToken, OHMRec, swapTarget, swapData );
```
@suhailgme mentioned something about getting price on-chain.

FinagleLord committed:
added _depositor to deposit function

FinagleLord committed:
added _depositor to deposit function

FinagleLord committed:
"cast depository to an address"

FinagleLord committed:
Update OlympusZap.sol

FinagleLord committed:
Update OlympusZapManager.sol

FinagleLord committed:
"approve the deposit to the bond depository"

Commit:
Create .gitignore

Commit:
Add missing import

Commit:
Add bondPrice function for convenience

Commit:
Update .gitignore

Commit:
Initial setup

Commit:
Create helper libs

- Adds addresses and other constants for use in tests
- Adds an exchange and approve helper
- Adds quote helpers to fetch quotes from 0x and Zapper APIs
- Adds a token helper for approvals and balance checks

Commit:
Add tracer for tests

Commit:
Transfer tokens to OlympusZapManager

- The Zap must grant the manager an allowance to spend the LP tokens. The manager must subsequently transfer these LP tokens from the Zap to itself before depositing into the depo

Commit:
Add tests

FinagleLord committed:
Manager removed

FinagleLord committed:
Manager combined, and OP support added

- extra param was needed on zapIn, bondPayoutToken, to support OP
- principalToBondDepository now takes two params, principal and bondPayoutToken, to support OP

FinagleLord committed:
Manager removed

Commit:
Merge branch 'main' into dev-env

FinagleLord committed:
typo fix

FinagleLord committed:
flattened

FinagleLord committed:
compile ready

FinagleLord committed:
compiler ready

FinagleLord committed:
merging suhails work

-I changed tests to work for the new logic, however I'm getting an error so it will still need a bit of work

FinagleLord committed:
Revert "merging suhails work"

This reverts commit c17ce35a4c76f9c3fa3a93137d87167a1e2e850f.

suhailgme committed:
Merge branch 'main' into dev-env

suhailgme committed:
Fix tests (clone from @FinagleLord)

suhailgme committed:
Update .gitignore

suhailgme committed:
Update settings.json

suhailgme committed:
update_BondDepos fixes

- Remove depos as approvedTargets as this is only used for contracts called by _fillQuote
- Use ZapBaseV2_2 _approveToken for pre-approval

FinagleLord committed:
Merge pull request #3 from suhailgme/dev-env

Setup Environment for Tests + Token Transfer Fix

FinagleLord committed:
Delete OlympusZapManager.sol
